### DISCUSSION TITLE
Not-linear interface kernel/ BC

### DISCUSSION THREAD

**Discussion by sid-mungale** [idaholab/moose#28456] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hi, I am trying to set up an interface between two blocks A and B that looks like flux = h(nonlinear var in A, nonlinear var in B).
The closest MOOSE objects I have seen are the InterfaceReaction and InterfaceDiffusion kernels, though as far as I can tell these do not allow arbitrary functions of the nonlinear variables on either side of the interface.

In a previous question I have seen test/src/materials/JumpInterfaceMaterial.C mentioned and have taken a look at it. Would I have to implement something similar to this? Or is there a MOOSE object I am missing? Any guidance here would be appreciated.

**Comment by GiudGiud** [comment, marked as answer]
Hello

InterfaceReaction and InterfaceDiffusion do not allow for an arbitrary function indeed, but they do allow for the dependence on variables on either side of the interface and are a good example to start from to create your own interface kernel that can handle an arbitrary h function.
Interface kernels are designed to work on interfaces between two subdomains, with a conformal mesh at the interface, and with dependencies on variables defined on one or the other side of the interface

Guillaume
